snap/validate.go

```diff
@@ -65,6 +68,27 @@ func isValidName(name string) bool {
 	return true
 }
 
+// ValidateName checks if a string can be used as a snap instance name.
+func ValidateInstanceName(instanceName string) error {
+	// NOTE: This function should be synchronized with the two other
+	// implementations: sc_instance_name_validate and validate_instance_name .
+	pos := strings.IndexRune(instanceName, '_')
+	if pos == -1 {
+		// just store name
+		return ValidateName(instanceName)
+	}
+
+	storeName := instanceName[0:pos]
+	instanceKey := instanceName[pos+1:]
+	if err := ValidateName(storeName); err != nil {
+		return err
+	}
+	if !validInstanceKey.MatchString(instanceKey) {
+		return fmt.Errorf("invalid snap instance key: %q", instanceKey)
```

Comment:
is the name 'instance key' being presented to the user in other ways, such that this isn't gibberish?

Reply:
No, this is probably the only place where it's presented without snap name, in other places it's always `yada yada snap_instance blah blah`. The whole 'instance name' becomes invalid because of instance key.